### PR TITLE
feat: refactor string helper to object

### PR DESCRIPTION
**Pull request by brendt**
Related to #288 and #430

### CODE REVIEW

**Comment by brendt** [comment]
So I wanted to add this object notation, but then thought the static variant isn't _really_ needed: we don't need two ways to do the same thing. Wdyt @innocenzi ?

File: composer.json [hunk 1]
```diff
@@ -94,7 +94,8 @@
             "src/Tempest/EventBus/src/functions.php",
             "src/Tempest/Http/src/functions.php",
             "src/Tempest/Mapper/src/functions.php",
-            "src/Tempest/Reflection/src/functions.php"
+            "src/Tempest/Reflection/src/functions.php",
+            "src/Tempest/Support/src/functions.php"
```
Comment (Treggats) [inline composer.json:98]: reduce the diff and overal convention (as far as I know)
```suggestion
            "src/Tempest/Support/src/functions.php",
```

File: src/Tempest/Support/composer.json [hunk 2]
```diff
@@ -11,7 +11,10 @@
     "autoload": {
         "psr-4": {
             "Tempest\\Support\\": "src"
-        }
+        },
+        "files": [
+            "src/functions.php"
```
Comment (Treggats) [inline src/Tempest/Support/composer.json:16]:
```suggestion
            "src/functions.php",
```

File: src/Tempest/Support/src/StringHelper.php [hunk 3]
```diff
@@ -1,79 +1,104 @@
 <?php
 
-declare(strict_types=1);
```
Comment (Treggats) [inline src/Tempest/Support/src/StringHelper.php]: Why'd you remove this?

File: src/Tempest/Support/src/StringHelper.php [hunk 4]
```diff
@@ -5,75 +5,105 @@
...
+use function preg_replace;
+use function rtrim;
+use Stringable;
+use function trim;
+
+final readonly class StringHelper implements Stringable
 {
-    public static function title(string $value): string
+    public function __construct(private string $string = '')
+    {
```
Comment (Treggats) [inline src/Tempest/Support/src/StringHelper.php:18]: This is a personal preference, do with it what you will 😄 
```suggestion
    public function __construct(
        private string $string = '',
    ) {
```

**Review by Treggats — CHANGES_REQUESTED** [review]

File: composer.json (hunk 1 above)
Comment (brendt) [reply composer.json:98]: Trailing commas aren't allowed in JSON :(

Comment (Treggats) [reply composer.json:98]: Oh right 🤦

**Review by Treggats — APPROVED** [review]
Looks good to me

**Comment by brendt** [comment]
Awesome, thanks!

**Pull request merged**